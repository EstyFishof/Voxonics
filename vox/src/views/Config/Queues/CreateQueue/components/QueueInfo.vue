<template>
  <div class="personal-info-wrap">
    <div class="wrap-head">
      <h2>QUEUE INFO</h2>
      <span class="hr"></span>
    </div>

    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <label>
              Name
              <input placeholder="Enter name of Queue"
                     class="input-field"
                     v-model.trim="name"
                     @change="setQueueName"
                     maxlength="12"
                     required
                     :class="{error: requiredFields.name}"
              />
            </label>
            <label>
              Number of Queue
              <input placeholder="Enter number of Queue"
                     class="input-field"
                     @keydown="onKeyDownHandlerForPriorityInput"
                     v-model.trim="internalNumber"
                     @change="setQueueNumber"
                     maxlength="4"
                     required
                     :class="{error: requiredFields.internalNumber}"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
    <div class="wrap-head options-title">
      <h2>Call Settings</h2>
      <span class="hr"></span>
    </div>
    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err" @click="hideListOptions(); indexRef='false'" v-click-outside="hide" ref="hideList">
            <label class="strategy-label options-label" :ref="'showStrategy'" @click="indexRef = 'showStrategy'" >
              Strategy
              <input class="input-field num" type="text" :value="newQueueStrategy" readonly>
              <span @click="setShowOptions('showStrategy')" class="triangle" :class="{'rotate-arrow': showOptions === 'showStrategy'}">
                <img src="../../../../../assets/img/icons/select-arrow.svg" alt="select">
              </span>
              <ul v-show="showOptions === 'showStrategy'" class="list-related-numbers">
                <li @click="selectOptions('strategy', 'Ring all')"
                :content="'Calls all users at the same time.'" v-tippy="{ placement: 'left', distance: '-35', arrow: true, delay: 50, duration: 500 }">Ring all</li>
                <li @click="selectOptions('strategy', 'Least recent')"
                :content="'Call the user who did not receive calls for the longest time.'" v-tippy="{ placement: 'left', distance: '-35', arrow: true, delay: 50, duration: 500 }">Least recent</li>
                <li @click="selectOptions('strategy', 'Fewest calls')"
                :content="'Call the user who received the least number of calls.'" v-tippy="{ placement: 'left', distance: '-35', arrow: true, delay: 50, duration: 500 }">Fewest calls</li>
                <li @click="selectOptions('strategy', 'Random')"
                :content="'Randomly distribute calls.'" v-tippy="{ placement: 'left', distance: '-35', arrow: true, delay: 50, duration: 500 }">Random</li>
                <li @click="selectOptions('strategy', 'RR memory')"
                :content="'Calling in a circle, after the agent who answered last.'" v-tippy="{ placement: 'left', distance: '-35', arrow: true, delay: 50, duration: 500 }">RR memory</li>
                <li @click="selectOptions('strategy', 'Linear')"
                :content="'Call starting from the first in the listed order.'" v-tippy="{ placement: 'left', distance: '-35', arrow: true, delay: 50, duration: 500 }">Linear</li>
              </ul>
            </label>
            <label class="strategy-label options-label" :ref="'showTimeout'" @click="indexRef = 'showTimeout'">
              Dial time
              <input class="input-field num" type="text" :value="newQueueTimeout" readonly>
              <span @click="setShowOptions('showTimeout')" class="triangle" :class="{'rotate-arrow': showOptions === 'showTimeout'}">
                <img src="../../../../../assets/img/icons/select-arrow.svg" alt="select">
              </span>
              <ul v-show="showOptions === 'showTimeout'" class="list-related-numbers">
                <li @click="selectOptions('timeout', `${index + 1}`)" v-for="(value, index) in new Array(60).fill(0)" :key="index">{{index + 1}}</li>
              </ul>
            </label>
            <label class="strategy-label options-label" :ref="'showRing'" @click="indexRef = 'showRing'">
              Ring in use
              <input class="input-field num" type="text" :value="newQueueRing" readonly>
              <span @click="setShowOptions('showRing')" class="triangle" :class="{'rotate-arrow': showOptions === 'showRing'}">
                <img src="../../../../../assets/img/icons/select-arrow.svg" alt="select">
              </span>
              <ul v-show="showOptions === 'showRing'" class="list-related-numbers">
                <li @click="selectOptions('ring', 'Yes')">Yes</li>
                <li @click="selectOptions('ring', 'No')">No</li>
              </ul>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import ClickOutside from "vue-click-outside"
import { clickCheck } from '@/lib/clickCheck.js'

export default {
  name: "QueueInfo",
  props: ['requiredFields'],
  data() {
    return {
      name: "",
      internalNumber: '',
      showOptions: '',
      strategy: '',
      timeout: '',
      ring: '',
      indexRef: ''
    }
  },
  computed: {
    ...mapGetters(['newQueue', 'newQueueName', 'newQueueInternalNumber', 'newQueueStrategy', 'newQueueTimeout', 'newQueueRing'])
  },
  methods: {
    onKeyDownHandlerForPriorityInput(evt) {
      if(!Number.isInteger(+evt.key) && !['Backspace', 'Delete'].includes(evt.key)) {
        evt.preventDefault()
      }
    },
    hideListOptions() {
      clickCheck(this, this.indexRef, this.$refs)
    },
    hide() {
      this.showOptions = ''
    },
    setQueueName() {
      this.$store.dispatch('setNameQueue', this.name)
    },
    setQueueNumber() {
      this.$store.dispatch('setInternalNumberQueue', this.internalNumber)
    },
    setShowOptions(option) {
      if(this.showOptions === option) {
        this.showOptions = ''
      } else {
        this.showOptions = option
      }
    },
    selectOptions(option, value) {
      switch (option) {
        case 'strategy':
            this.$store.dispatch('setStrategyQueue', value)
          break
        case 'timeout':
            this.$store.dispatch('setTimeoutQueue', value)
          break
        case 'ring':
            this.$store.dispatch('setRingQueue', value)
          break
        default:
          break
      }
      this.showOptions = ''
    }
  },
  mounted() {
    this.popupItem = this.$refs.hideList
    this.$store.dispatch('setStrategyQueue', 'Ring all')
    this.$store.dispatch('setTimeoutQueue', '21')
    this.$store.dispatch('setRingQueue', 'Yes')

    if (this.newQueueName) {
      this.name = this.newQueueName
    }
    if(this.newQueueInternalNumber) {
      this.internalNumber = this.newQueueInternalNumber
    }
    if(this.newQueueStrategy) {
      this.strategy = this.newQueueStrategy
    }
    if(this.newQueueTimeout) {
      this.timeout = this.newQueueTimeout
    }
    if(this.newQueueRing) {
      this.ring = this.newQueueRing
    }
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss">
.personal-info-wrap {
  .options-title {
    margin-left: 24px;
    h2 {
      font-weight: bold;
      font-size: 32px;
      line-height: 37px;
    }
  }
  .wrap-body {
    display: flex;
    flex-direction: row;
    margin: 36px 0 36px 24px;

    .personal-info {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;

      &__first-part {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;

        .item-container {
          display: flex;
          margin: 16px 0;

          label {
            font-size: 18px;
            font-weight: 700;
            text-align: left;
            color: var(--light-blue);
            margin-right: 48px;

            .triangle {
              width: calc(104% - 7px);
              margin-bottom: 4px;
            }

            .input-field {
              padding-left: 12px;
            }

            .question {
              width: 10px;
              margin-left: -5px;
              margin-bottom: 8px;
            }

            .error {
              border: 1px solid red;
              outline: red;
            }
          }

          .question {
            width: 10px;
            margin-left: -5px;
            margin-bottom: 8px;
          }

          .options-label {
            font-size: 18px;
            font-weight: 700;
            text-align: left;
            color: var(--light-blue);
            margin-right: 48px;
            position: relative;

            .input-field {
              margin-top: 12px;
              padding: 16px 8px;
              width: 292px;
              height: 35px;
              background: var(--permission-table-bg);
              color: var(--font-panel);
              font-weight: 300;
              padding-left: 12px;
            }

            .triangle {
              img {
                margin-right: 10px;
              }
            }

            .list-related-numbers{
              max-height: 150px;
              overflow: auto;
              box-shadow: 0 8px 16px rgba(0, 0, 0, 0.12);
                li {
                  height: 35px;
                  font-weight: 300;
                  padding: 8px 16px;
                }
              }
            }
        }
      }
    }
  }

  .with-err {
    position: relative;
    margin-bottom: 30px;
  }

}
@media screen and (max-width: 1724px) {
  .personal-info-wrap {
    .wrap-body {
      .personal-info__first-part {
        .item-container {
          .options-label {
            .input-field {
              width: 260px;
            }
          }
        }
      }
    }
  }
}
</style>
