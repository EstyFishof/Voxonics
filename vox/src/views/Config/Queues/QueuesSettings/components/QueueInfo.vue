<template>
  <div class="personal-info-wrap" :class="{'disabled-events': !$access.isEditDepAndQueue(userInfo, 'queue', getQueueInfo)}">
    <div class="wrap-head">
      <h2>QUEUE INFO</h2>
      <span class="hr"></span>
      <img @click="del()" class="trash" src="@/assets/img/icons/trash.svg" alt="trash" width="30" :class="{'grayscale': !$access.isEditDepAndQueue(userInfo, 'queue', getQueueInfo)}">
    </div>

    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <span>  Name of Queue:</span>
          <div class="item-container with-err" :class="{'grayscale': !$access.isEditDepAndQueue(userInfo, 'queue', getQueueInfo)}">
            <div class="left-section-block">
              <input :style="{width: computedWidth}"
                    placeholder="Enter name of Queue"
                    :class="{'input-field': edit}"
                    :defaulValue="getQueueNameInfo"
                    v-model.trim="name"
                    maxlength="12"
                    @input="setQueueName"
              />
              <span class="last-update-block">
                <span class="last-update-text">Last updated:</span>
                <span class="last-update-data">{{getQueueInfo.dateUpdated}}</span>
              </span>
            </div>
            <div class="wrap-static">
              <div class="static">
                <div class="label">Number of Queue:</div>
                <div class="num">{{ getQueueInfo.internalNumber }}</div>
              </div>
              <div class="static">
                <div class="label">Agents in Queue:</div>
                <div class="num">{{ getAgentsLength || 0 }}</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import store from "../../../../../store"

export default {
  name: "QueueInfo",
  data() {
    return {
      name: "",
      edit: false
    }
  },
  computed: {
    ...mapGetters(['numbersDep', 'getAgentsQueueInfo', 'managersDep', 'getQueueNameInfo', 'getQueueInfo', 'getAgentsLength', 'userInfo', 'getQueueInfo']),
    computedWidth() {
      return this.name.length ? 32 + this.name.length * 1.1 * 24 + 'px' : '100%'
    }
  },
  methods: {
    del() {
      this.$store.dispatch('setDeleteQueueInfo', true)
    },
    setQueueName() {
      this.$store.dispatch('setQueueNameInfo', this.name)
    },
  },
  mounted() {
    this.name = this.getQueueNameInfo
    store.subscribe((mutation) => {
      if (mutation.type === 'queueInfoMutation') {
        this.name = this.getQueueNameInfo
      }
    })
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/departmentsAndQueues/info.scss"></style>
