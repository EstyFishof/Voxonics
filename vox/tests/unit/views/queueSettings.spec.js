import {
    shallowMount,
    createLocalVue
} from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import QueueSetting from '../../../src/views/Config/Queues/QueuesSettings/QueuesSettings.vue';
import AgentsSetting from '../../../src/views/Config/Queues/QueuesSettings/components/AgentsSettings.vue';
import ConfigSetting from '../../../src/views/Config/Queues/QueuesSettings/components/ConfigSettings.vue';
import DeleteQueue from '../../../src/views/Config/Queues/QueuesSettings/components/DeleteQueue.vue';
import ManagersSetting from '../../../src/views/Config/Queues/QueuesSettings/components/ManagersSettings.vue';
import QueueInfo from '../../../src/views/Config/Queues/QueuesSettings/components/QueueInfo.vue';
import VueTippy, { TippyComponent } from "vue-tippy";


const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);
const router = new VueRouter();
localVue.use(Vuex);

describe('QueueSetting', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            queueInfo: {
                uid: 5000,
                strategy: 'Random',
                ringinuse: 'ringinuse',
                relatedWith: {user:[]},
                belongTo: []
            },
            deleteQueue: {}
        },
        mutations: {
            queueInfoMutation(state, value) {
                state.queueInfo = value;
            }
        },
        getters: {
            getQueueInfo(state) {
                return state.queueInfo;
            },
        }
    });
    const wrapper = shallowMount(QueueSetting, {
        store,
        localVue,
        router,
        stubs: {QueueSetting: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    // it('Check snapshot', () => {
    //     expect(wrapper.element).toMatchSnapshot();
    // })
    it('setShowOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setShowOptions');
        await wrapper.vm.setShowOptions('config');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('handlerSubmit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'handlerSubmit');
        await wrapper.vm.handlerSubmit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('clearDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'clearDepartment');
        await wrapper.vm.clearDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('mounted component', () => {
        QueueSetting.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        store.commit('queueInfoMutation', {
            strategy: 'Random',
            ringinuse: 'ringinuse',
            relatedWith: [],
            belongTo: []
        });
        await QueueSetting.mounted.call(wrapper.vm);
    });
});

describe('AgentsSetting', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            agents: [],
            queueAgetns: [],
            agentsInfo: {
                user: []
            }
        },
        actions: {
            setAgentsQueueInfo({
                commit
            }, value) {}
        },
        mutations: {
            queueInfoMutation(state, value) {
                state.queueInfo = value;
            }
        },
        getters: {
            agentsQueue(state) {
                return state.agents;
            },
            agentsDep(state) {
                return state.queueAgetns;
            },
            getAgentsQueueInfo(state) {
                return state.agentsInfo;
            },
        }
    });
    const wrapper = shallowMount(AgentsSetting, {
        store,
        localVue,
        router,
        stubs: {AgentsSetting: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await wrapper.vm.chooseAgent({
            uid: 111
        });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        wrapper.setData({
            relatedAgents: [{
                uid: 111
            }]
        });
        await wrapper.vm.chooseAgent({
            uid: 111
        });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setAgentsDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setAgentsDepartment');
        await wrapper.vm.setAgentsDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('computed agentsFiltered', () => {
    //     AgentsSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({
    //         name: '123'
    //     })
    //     AgentsSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({
    //         name: '123'
    //     })
    //     AgentsSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({
    //         name: 'ффф'
    //     })
    //     AgentsSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    it('mounted component', () => {
        AgentsSetting.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        store.commit('queueInfoMutation', {
            strategy: 'Random',
            ringinuse: 'ringinuse',
            relatedWith: [],
            belongTo: []
        });
        await AgentsSetting.mounted.call(wrapper.vm);
    });
});

describe('ConfigSetting', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            strategy: '',
            timeout: '',
            ringInUse: '',
            wait: ''
        },
        actions: {
            setStrategyInfo({
                commit
            }, value) {},
            setTimeoutInfo({
                commit
            }, value) {},
            setRingInUseInfo({
                commit
            }, value) {}
        },
        getters: {
            getStrategyInfo(state) {
                return state.strategy;
            },
            getTimeoutInfo(state) {
                return state.timeout;
            },
            getRingInUseInfo(state) {
                return state.ringInUse;
            },
            getMaxWaitInfo(state) {
                return state.wait;
            },
        }
    });
    const wrapper = shallowMount(ConfigSetting, {
        store,
        localVue,
        router,
        stubs: {ConfigSetting: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('setShowOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setShowOptions');
        await wrapper.vm.setShowOptions('showRing');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'selectOptions');
        await wrapper.vm.selectOptions('strategy', 'ring all');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'selectOptions');
        await wrapper.vm.selectOptions('timeout', '10');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'selectOptions');
        await wrapper.vm.selectOptions('ring', 'yes');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'selectOptions');
        await wrapper.vm.selectOptions('', 'yes');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });

});

describe('DeleteQueue', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            deleteQueue: {}
        },
        actions: {
            setDeleteQueueInfo({
                commit
            }, value) {},
        },
        getters: {
            getDeleteQueueInfo(state) {
                return state.deleteQueue;
            },
        }
    });
    const wrapper = shallowMount(DeleteQueue, {
        store,
        localVue,
        router,
        stubs: {DeleteQueue: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('cancelDelete toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'cancelDelete');
        await wrapper.vm.cancelDelete();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('deleteQueue toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'deleteQueue');
        await wrapper.vm.deleteQueue();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});

describe('ManagersSetting', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            managersQ: [],
            managerQInfo: {
                user: [
                    {
                        uid: '',
                        option: {
                            edit: 1
                        }
                    }
                ],
            },
            QLength: 1
        },
        actions: {
            setMagagersQueueInfo({
                commit
            }, value) {},
        },
        getters: {
            managersQueue(state) {
                return state.managersQ;
            },
            getManagersQueueInfo(state) {
                return state.managerQInfo;
            },
            getAgentsLength(state) {
                return state.QLength;
            }
        }
    });
    const wrapper = shallowMount(ManagersSetting, {
        store,
        localVue,
        router,
        stubs: {ManagersSetting: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        wrapper.setData({relatedManagers: [{uid: 222, option: {edit: true}}]});
        await wrapper.vm.chooseAgent({uid: 111, option: {edit: true}});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        wrapper.setData({relatedManagers: [{uid: 111, option: {edit: true}}]});
        await wrapper.vm.chooseAgent({uid: 111, option: {edit: true}});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('changeEdit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeEdit');
        await wrapper.vm.changeEdit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setAgentsDepartment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setAgentsDepartment');
        await wrapper.vm.setAgentsDepartment();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('agentsFiltered agentsFiltered', async () => {
    //     await ManagersSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('agentsFiltered agentsFiltered', async () => {
    //     wrapper.setData({name: '123'})
    //     await ManagersSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('agentsFiltered agentsFiltered', async () => {
    //     wrapper.setData({name: 'aaa'})
    //     await ManagersSetting.computed.agentsFiltered.call(wrapper.vm)
    // })
    it('mounted component', async () => {
        await ManagersSetting.mounted.call(wrapper.vm);
    });
});

describe('QueueInfo', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            nameQ: '',
            agents: {
                user: ''
            },
            infoQ: {
                internalNumber: 1111
            },
            agentsLength: 4
        },
        actions: {
            setDeleteQueueInfo({
                commit
            }, value) {},
            setQueueNameInfo({
                commit
            }, value) {},
        },
        getters: {
            getQueueNameInfo(state) {
                return state.nameQ;
            },
            numbersDep(state) {
                return state.nameQ;
            },
            getAgentsQueueInfo(state) {
                return state.agents;
            },
            managersDep(state) {
                return state.nameQ;
            },
            getQueueInfo(state) {
                return state.infoQ;
            },
            getAgentsLength(state) {
                return state.agentsLength;
            }
        }
    });
    const wrapper = shallowMount(QueueInfo, {
        store,
        localVue,
        router,
        stubs: {QueueInfo: true}
    });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('del toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'del');
        await wrapper.vm.del();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setQueueName toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setQueueName');
        await wrapper.vm.setQueueName();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('toggleEdit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'toggleEdit');
        await wrapper.vm.toggleEdit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('agentsFiltered agentsFiltered', async () => {
        await QueueInfo.computed.computedWidth.call(wrapper.vm);
    });
    it('mounted component', async () => {
        await QueueInfo.mounted.call(wrapper.vm);
    });

});
