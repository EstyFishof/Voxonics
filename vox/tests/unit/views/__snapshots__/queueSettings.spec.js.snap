// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentsSetting Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      Agents in Queue
    </h2>
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Add agent

    </div>
     
    <div
      class="agents users"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter name or Agent Number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="agents users"
    >
      <drop-list-stub
        acceptsdata="function () { return true; }"
        acceptstype="function () { return true; }"
        dragimageopacity="0.7"
        items=""
        mode="cut"
        tag="div"
      />
    </div>
  </div>
</div>
`;

exports[`ConfigSetting Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      Settings
    </h2>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <label
            class="strategy-label options-label"
          >
            
              Strategy
 
            <input
              class="input-field num"
              readonly="readonly"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="../../../../../assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                Ring all
              </li>
               
              <li>
                Least recent
              </li>
               
              <li>
                Fewest calls
              </li>
               
              <li>
                Random
              </li>
               
              <li>
                RR memory
              </li>
               
              <li>
                Linear
              </li>
               
              <li>
                W random
              </li>
            </ul>
          </label>
           
          <label
            class="strategy-label options-label"
          >
            
              Timeout
 
            <input
              class="input-field num"
              readonly="readonly"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="../../../../../assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                0
              </li>
               
              <li>
                1
              </li>
               
              <li>
                2
              </li>
               
              <li>
                3
              </li>
               
              <li>
                4
              </li>
               
              <li>
                5
              </li>
            </ul>
          </label>
           
          <label
            class="strategy-label options-label"
          >
            
              Ring in use
 
            <input
              class="input-field num"
              readonly="readonly"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="../../../../../assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                Yes
              </li>
               
              <li>
                No
              </li>
            </ul>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DeleteQueue Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="delete-block"
  >
    <div
      class="delete-top-block"
    >
      <span
        class="delete-image"
      >
        <img
          alt="delete"
          src="@/assets/img/icons/delete-queue.svg"
        />
      </span>
       
      <div
        class="top-content-block"
      >
        <span
          class="top-title"
        >
          Delete Queue?
        </span>
         
        <span
          class="top-content"
        >
          Are you sure delete this Queue 
        </span>
      </div>
    </div>
     
    <div
      class="delete-btn-block"
    >
      <span
        class="cancel-btn"
      >
        Cancel
      </span>
       
      <span
        class="delete-btn"
      >
        Delete 
      </span>
    </div>
  </div>
</div>
`;

exports[`ManagersSetting Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      Managers in Department
    </h2>
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Add manager
      
    </div>
     
    <div
      class="managers users"
    >
      <div
        class="manager user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            Name
          </div>
           
          <span
            class="name"
          />
        </div>
         
        <span
          class="name"
        >
          Agent Number
        </span>
         
        <span
          class="union"
        />
      </div>
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter name or Agent Number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="managers users"
    >
      <div
        class="manager user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            <img
              alt="avatar"
              src=""
            />
          </div>
           
          <span
            class="name"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <div
          class="action"
        >
          <label
            class="label-container"
          >
            <input
              class="edit"
              type="checkbox"
            />
             
            <span
              class="check"
            />
             
            <span
              class="text"
            >
              Edit
            </span>
          </label>
        </div>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`QueueInfo Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      QUEUE INFO
    </h2>
     
    <span
      class="hr"
    />
     
    <img
      alt="trash"
      class="trash"
      src="@/assets/img/icons/trash.svg"
      width="30"
    />
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <span>
            Name of Queue:
        </span>
         
        <div
          class="item-container with-err"
        >
          <input
            class=""
            defaulvalue=""
            maxlength="12"
            placeholder="Enter name of Queue"
            style="width: 100%;"
          />
           
          <div
            class="wrap-static"
          >
            <div
              class="static"
            >
              <div
                class="label"
              >
                Number of Queue:
              </div>
               
              <div
                class="num"
              >
                1111
              </div>
            </div>
             
            <div
              class="static"
            >
              <div
                class="label"
              >
                Agents in Queue:
              </div>
               
              <div
                class="num"
              >
                4
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
